pkg/controller/v1beta1/inferenceservice/utils/utils.go

```diff
@@ -208,6 +208,16 @@ func MergeRuntimeContainers(runtimeContainer *v1.Container, predictorContainer *
 
 	// Strategic merge patch will replace args but more useful behaviour here is to concatenate
 	mergedContainer.Args = append(append([]string{}, runtimeContainer.Args...), predictorContainer.Args...)
+	// Remove duplicates
+	seenArgs := make(map[string]bool)
+	var result []string
+	for _, arg := range mergedContainer.Args {
+		if _, ok := seenArgs[arg]; !ok {
+			seenArgs[arg] = true
+			result = append(result, arg)
```

Comment:
What happens for the case that key and value are separated to different args ?

For example
—foo
dummy
—foo
test

Reply:
I realized that the current args from two container specs are being merged in a naive way and cannot handle complex use cases.
1. Key and value are separated to different args, e.g. `--foo bar`  vs `--foo=newBar`
1. Boolean flags, especially when it comes to merging/overriding, e.g. `--bool-flag`
1. Handle duplicates and different ordering of args

The code can be complex if we want to cover these cases.

Instead of overriding individual args, we should simply replace the entire list of args by the list of args defined in `InferenceService`.

I have just updated the PR based on the new approach.